woocommerce/woocommerce
# Storybook: Clean up dependencies

### All Submissions:






### Changes proposed in this Pull Request:

[Storybook was added to `tools/` recently](https://github.com/woocommerce/woocommerce/pull/32729) and its dependencies were added to the top level `package.json`. This wasn't great for keeping things tidy in the top level `package.json`. This PR cleans things up by designating `tools/storybook` as its own project so its dependencies can be declared within the new project.

In moving dependencies, errors were being thrown because certain packages were missing undeclared dependencies that previously worked because they were declared elsewhere. Those dependencies were added in packages that required them.

Also, since the scripts were moved to the new package as well, an NX command was also added. Seeing as we will migrate from NX to Turbo soon, this is merely a placeholder.

### How to test the changes in this Pull Request:

1. `pnpm install`
2. `pnpm nx watch storybook`
3. See Storybook load in the browser without error.
4. Note, if you see an error in the browser similar to [this one](https://github.com/storybookjs/storybook/issues/10887), remove `node_modules`, `tools/storybook/node_modules`, and `pnpm-lock.yaml` and start over. For me there was an error involving mismatched modules.

### Other information:




### FOR PR REVIEWER ONLY:
